bin/thumbnail

```diff
@@ -285,6 +285,49 @@ function set_custom_type() {
 	OUT="${1}:${OUT}"
 }
 
+function recalculate_requested_size_for_crop_down() {
+	# recalculate output image dimensions in cases of "...-crop-down" modes
+	# in such a way that an output image has the same aspect ratio as request
+	# after executing imagemagick
+	local origwidth=$1
+	local origheight=$2
+	local CODE
+	local x
+	local y
+
+	# need to use bc for floating-point arithmetic below
+	local CODE=`cat <<EOF
+scale = 20
+define i(x) {
+    auto s
+    s = scale
+    scale = 0
+    x /= 1   /* round x down */
+    scale = s
+    return (x)
+}
+
+s = 1
+u = $origwidth / $WIDTH
+if (u<s) s = u
+
+u = $origheight / $HEIGHT
+if (u<s) s = u
+
+i($WIDTH * s + 0.000000001)
+i($HEIGHT * s + 0.000000001)
```

Comment:
This is required due to rounding issues that are common in floating-point operations. Moreover there were a lot of cases where the value of ``$WIDTH * s`` resulted in eg. ``599.99999999999235825`` which should become ``600``. The value of ``0.000000001`` is often described as ``epsilon`` which you use just before truncating the fraction point so you get results closer to what you wanted.